### DISCUSSION TITLE
float pydantic schema auto truncate/round

### DISCUSSION THREAD

**Discussion by yuuuxt** [fastapi/fastapi#7712] [Questions] [answered]
Is it possible to have a pydantic schema that could auto truncate/round the output?

for example, if I have a `class FloatResult(BaseModel)` that has `result: float`, the json output of `1 / 3` would be `{"result": 0.3333333333333333}`, but I may want an output of `{"result": 0.33333}`.

How could I configure it, so that I can specify the length of output easily (I hope it only affects the json output)?

**Comment by Dustyposa** [comment]
You can try `Validators`, like this:
```python
from pydantic import BaseModel, validator


class FloatResult(BaseModel):
    result: float
    
    @validator('result')
    def result_check(cls, v):
        ...
        return round(v, 5)
```

For more information, [here](https://pydantic-docs.helpmanual.io/usage/validators/)

**Reply by sshishov** [reply]
@tiangolo @Dustyposa

If I am not mistaken, it will round the value inside, and not only output. It means if inside you have some math in future, you will work with rounded value. I guess it is not the proper solution.

The proper solution should somehow keep the value not rounded inside the model itself but when you use it or print, it will be rounded...

**Reply by YuriiMotov** [reply, marked as answer]
```python
from pydantic import BaseModel, field_serializer

class FloatResult(BaseModel):
    result: float
    
    @field_serializer('result', mode="plain")
    @classmethod
    def round_val(cls, v: float):
        return str(round(v, 5))
```

**Comment by yuuuxt** [comment]
> 
> 
> You can try `Validators`, like this:
> 
> ```python
> from pydantic import BaseModel, validator
> 
> 
> class FloatResult(BaseModel):
>     result: float
>     
>     @validator('result')
>     def result_check(cls, v):
>         ...
>         return round(v, 5)
> ```
> 
> For more information, [here](https://pydantic-docs.helpmanual.io/usage/validators/)

It works as expected. Thanks!

**Comment by tiangolo** [comment]
Thanks for the help here @Dustyposa ! :clap: :bow: 

Thanks for reporting back and closing the issue @yuuuxt  :+1:
